app/src/main/java/com/android/solvit/shared/model/provider/Schedule.kt

```diff
@@ -131,37 +131,36 @@ data class Schedule(
   fun isAvailable(dateTime: LocalDateTime): Boolean {
     val time = dateTime.toLocalTime()
 
-    // Check for exceptions first
+    // Check for off-time exceptions first, as they override all other availability
     val exception = exceptions.find { it.date.toLocalDate() == dateTime.toLocalDate() }
-    if (exception != null) {
-      when (exception.type) {
-        ExceptionType.OFF_TIME -> {
-          // If there's an off-time exception overlapping with the time, provider is not available
-          if (exception.timeSlots.any { slot ->
-            (time.equals(slot.start) || time.isAfter(slot.start)) &&
-                (time.equals(slot.end) || time.isBefore(slot.end))
-          }) {
-            return false
-          }
-        }
-        ExceptionType.EXTRA_TIME -> {
-          // If there's an extra-time exception overlapping with the time, provider is available
-          if (exception.timeSlots.any { slot ->
-            (time.equals(slot.start) || time.isAfter(slot.start)) &&
-                (time.equals(slot.end) || time.isBefore(slot.end))
-          }) {
-            return true
-          }
-        }
+    if (exception?.type == ExceptionType.OFF_TIME) {
+      // Not available if there's an overlapping off-time slot
+      if (exception.timeSlots.any { slot ->
+        !time.isBefore(slot.start) && time.isBefore(slot.end)
```

Comment:
If I have a slot between 10:11 would 9:10 and 11:12 be available?

Reply:
Good catch! I forgot to use the overlaps function to allow perfect time slot overlap. Should now be fixed!